### PR TITLE
Stale agents improvements

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: src/Agent.php [hunk 1]
```diff
@@ -821,31 +821,56 @@ public static function cronCleanoldagents($task = null)
...
+                        }
+                        break;
+                    case Conf::STALE_AGENT_ACTION_TRASHBIN:
+                        //change status of agents linked assets
+                        foreach ($iterator as $data) {
+                            $itemtype = $data['itemtype'];
+                            if (is_subclass_of($itemtype, CommonDBTM::class)) {
+                                $item = new $itemtype();
+                                if ($item->getFromDB($data['items_id'])) {
+                                    $item->update([
```
Comment (cedric-anne) [inline src/Agent.php:863]: 1. Should item be updated even if it does not have the `is_dynamic=1` property ?
2. Why using `update` instead of `delete` ?

Comment (trasher) [reply src/Agent.php:863]: Indeed I should have used `delete()`, yes, but this is linked to the other question:  updating the `is_dynamic` field... Well, I set that to be consistant with state change (the same question/answer should apply).

File: install/migrations/update_10.0.5_to_10.0.6/inventory.php [hunk 2]
```diff
@@ -38,3 +38,16 @@
  * @var Migration $migration
  */
 $migration->addConfig(["entities_id_default" => 0], 'inventory');
+
+$config = \Config::getConfigurationValues('inventory');
+//convert stale_agents_action to an array
+$DB->update(
```
[the remark this answers is not in the document]
Comment (trasher) [reply install/migrations/update_10.0.5_to_10.0.6/inventory.php:44]: Yes, I'm not sure how to check if this is an array or not already.

File: src/Agent.php (hunk 1 above)
Comment (cedric-anne) [reply src/Agent.php:863]: My first question was : "Should we preserve items that are not marked as dynamic ?". I do not know if it is possible to have a computer linked to an agent that is not marked as dynamic anymore.

File: install/migrations/update_10.0.5_to_10.0.6/inventory.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (cedric-anne) [reply install/migrations/update_10.0.5_to_10.0.6/inventory.php:44]: As it should be a numeric value if it has not been updated yet, you can puth the update inside a `if (is_numeric($config['stale_agents_action'])) { }` block.

File: src/Agent.php (hunk 1 above)
Comment (trasher) [reply src/Agent.php:863]: In my mind, if an item is linked to an agent, then it's dynamic. I have no idea why the status change sets it explicitly...

We have two choices I guess: either force `is_dynamic` to 1 or check if it's 1 already (and if not, do not apply the change).

**Review by orthagh — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**